### PR TITLE
MEN-5154 Adding azure-iot-manager.

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:title
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: mender/values.yaml [hunk 1]
```diff
@@ -280,6 +280,29 @@ auditlogs:
...
+      cpu: 50m
+      memory: 128M
+    requests:
+      cpu: 50m
+      memory: 128M
+  affinity: {}
+  image:
+    registry: registry.mender.io
+    repository: mendersoftware/azure-iot-manager
+    tag: mender-3.1.0
```
Comment (tranchitella) [inline mender/values.yaml:298]: Uhm, this won't work until we release... shall we use mender-master in the meanwhile?
Maybe we should replace all the image tags in the master branch of the Helm chart to mender-master?

File: mender/templates/api-gateway-deploy.yaml [hunk 2]
```diff
@@ -42,7 +42,7 @@ spec:
       containers:
       - name: api-gateway
         image: {{ .Values.api_gateway.image.registry }}/{{ .Values.api_gateway.image.repository }}:{{ .Values.api_gateway.image.tag }}
-        imagePullPolicy: {{ .Values.useradm.image.imagePullPolicy }}
+        imagePullPolicy: {{ .Values.api_gateway.image.imagePullPolicy }}
```
Comment (tranchitella) [inline mender/templates/api-gateway-deploy.yaml:45]: Ooops, thanks for fixing these copy-and-paste errors!

File: mender/values.yaml (hunk 1 above)
Comment (merlin-northern) [reply mender/values.yaml]: changed to mender-master. I have no problem in using mender-master everywhere.

**Comment by tranchitella** [comment]
@merlin-northern yes, please, let's move everything to mender-master with an independent commit. It doesn't make sense to run azure-iot-manager with non-master images anyway, as they don't have the end-points required to fully integrate the service.

File: mender/templates/azure-iot-manager-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
+{{- if and .Values.azure_iot_manager.enabled .Values.global.enterprise }}
```
Comment (tranchitella) [inline mender/templates/azure-iot-manager-deploy.yaml:1]:
```suggestion
{{- if .Values.azure_iot_manager.enabled }}
```

File: README.md [hunk 4]
```diff
@@ -451,6 +451,33 @@ The following table lists the parameters for the `auditlogs` component and their
...
+
+### Parameters: azure-iot-manager
+
+The following table lists the parameters for the `azure-iot-manager` component and their default values:
+
+| Parameter | Description | Default |
+| -------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------- |
+| `azure_iot_manager.enabled` | Enable the component | `true` |
+| `azure_iot_manager.automigrate` | Enable automatic database migrations at service start up | `true` |
+| `azure_iot_manager.image.registry` | Docker image registry | `registry.mender.io` |
```
Comment (tranchitella) [inline README.md:463]:
```suggestion
| `azure_iot_manager.image.registry` | Docker image registry | `docker.io` |
```

File: mender/values.yaml [hunk 5]
```diff
@@ -280,6 +280,29 @@ auditlogs:
...
+  resources:
+    limits:
+      cpu: 50m
+      memory: 128M
+    requests:
+      cpu: 50m
+      memory: 128M
+  affinity: {}
+  image:
+    registry: registry.mender.io
```
Comment (tranchitella) [inline mender/values.yaml:296]:
```suggestion
    registry: docker.io
```

**Review by tranchitella — CHANGES_REQUESTED** [review]

File: mender/templates/azure-iot-manager-deploy.yaml (hunk 3 above)
Comment (merlin-northern) [reply mender/templates/azure-iot-manager-deploy.yaml]: fixed.

File: README.md (hunk 4 above)
Comment (merlin-northern) [reply README.md]: fixed.

File: mender/values.yaml (hunk 5 above)
Comment (merlin-northern) [reply mender/values.yaml]: fixed.

**Comment by merlin-northern** [comment]
> @merlin-northern yes, please, let's move everything to mender-master with an independent commit. It doesn't make sense to run azure-iot-manager with non-master images anyway, as they don't have the end-points required to fully integrate the service.

of course: https://github.com/mendersoftware/mender-helm/pull/60/commits/2069a5d295523e6b57fb56e5f546f83c7f75b149

**Review by tranchitella — APPROVED** [review]
@merlin-northern one final touch, please replace 3.1.0 with master (both places) here as well:
https://github.com/mendersoftware/mender-helm/blob/master/mender/Chart.yaml#L2

Other than this, it's okay!

**Comment by merlin-northern** [comment]
> @merlin-northern one final touch, please replace 3.1.0 with master (both places) here as well: https://github.com/mendersoftware/mender-helm/blob/master/mender/Chart.yaml#L2

like this: https://github.com/mendersoftware/mender-helm/compare/2069a5d295523e6b57fb56e5f546f83c7f75b149..1841fbadcfedc121afbf614432f4c3d20b9d32d7 ?

> Apart from this, it's okay!

thank you.

**Review by tranchitella — APPROVED** [review]

**Pull request merged**